### ISSUE TITLE
getLocation() never return if user don't accept to use Google location service

### ISSUE THREAD

**Issue by emersonsiega** [Lyokone/flutterlocation#210]
**Describe the bug**
On Android, when user is prompted to enable Google location service, if say "No Thanks", the getLocation method hangs and never return. No error is throw.

**Expected behavior**
Must return the location or some error if it's not possible.

**Tested on:**
 - Android, API Level 28, simulator and Moto G device
 - iOS, Version 12, simulator and real device (SE). Works fine

**Additional logs**
 - There's just one log that would indicate something...

``` shell
W/System  (15921): A resource failed to call release.
```

**Comment by mfarizon** [comment]
This issue should be kept open, as this is still a problem.
On Android, if you deactivate the localisation globally for the phone and authorize it for the app, the getLocation method hangs and never return. No error is throw.

**Comment by aleksandr-m** [comment]
Workaround:

```
bool getloc = false;
if (await _location.serviceEnabled()) {
  getloc = true;
} else {
  getloc = await _location.requestService();
}
if (getloc) {
  currentLocation = await _location.getLocation();
}
```

**Comment by hmmdeif** [comment]
Actually if you enter Yes it still never returns. The service can already be enabled, it's an additional option when using mobile data rather than Wifi so that the device can use nearby wifi to increase location accuracy.

**Comment by nicowernli** [comment]
@awhitford I believe it's unrelated, I'm having the problem in iOS

**Comment by srburton** [comment]
Escrevi outra, #263   também com o mesmo problema.

**Comment by JaMoLpE88** [comment]
The latest update was published on Apr 10, 2019. Are these dependencies still supported and are they planned to include the latest updates and solutions?  We have been waiting for this solution for many months and i can see that nicowernli solved It on #267.  Thanks.

**Comment by hmmdeif** [comment]
Owner is AWOL, it's probably best if @nicowernli or someone else active re-publishes this on pub.

**Comment by aebmknrl** [comment]
Same problem here. If the user hits ALLOW to permissions, then the execution of the code never returns.

**Comment by Lyokone** [comment]
Hey ! Should be fixed with 2.5.0, please reopen if this still happens :) Thanks for your patience :)
